Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -381,7 +381,7 @@ static bool margins_collapse_through(Box const& box, LayoutState& state)
     // nor top or bottom padding, and it has a 'height' of either 0 or 'auto', and it does not contain a line box, and
     // all of its in-flow children's margins (if any) collapse.
     // https://www.w3.org/TR/CSS22/box.html#collapsing-margins
-    return state.get(box).border_box_height() == 0;
+    return box.computed_values().clear() == CSS::Clear::None && state.get(box).border_box_height() == 0;
```

Comment:
hm, I don't think it's right way to solve the issue. will have to collect more test cases of how margins collapsing happens when there are boxes with clearance because I don't really understand clearance part of spec at this point.

Reply:
Is this comment still relevant?

Reply:
yes, unfortunately I haven't made any improvements since I left this comment.